ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java

```diff
@@ -547,26 +561,111 @@ private void importDocumentFromIntent(final Intent data) {
+                                @Override
+                                public void documentAccepted() {
+                                    LOG.debug("Client accepted the document");
+                                    hideActivityIndicatorAndEnableInteraction();
+                                    mListener.onDocumentAvailable(document);
+                                }
+
+                                @Override
+                                public void documentRejected(@NonNull final String messageForUser) {
+                                    LOG.debug("Client rejected the document: {}", messageForUser);
+                                    hideActivityIndicatorAndEnableInteraction();
+                                    showInvalidFileAlert(messageForUser);
+                                }
+                            });
+                }
+
+                @Override
+                public void onError(final Exception exception) {
+                    LOG.error("Failed to load document data", exception);
+                    hideActivityIndicatorAndEnableInteraction();
+                    showInvalidFileError(null);
+                }
+            });
         } catch (IllegalArgumentException e) {
             LOG.error("Failed to import selected document", e);
+            hideActivityIndicatorAndEnableInteraction();
             showInvalidFileError(null);
         }
     }
 
+    private void showActivityIndicatorAndDisableInteraction() {
+        if (mActivityIndicator == null
+                || mActivityIndicatorBackground == null) {
+            return;
+        }
+        mActivityIndicatorBackground.setVisibility(View.VISIBLE);
+        mActivityIndicatorBackground.setClickable(true);
+        mActivityIndicator.setVisibility(View.VISIBLE);
+        disableInteraction();
+    }
+
+    private void hideActivityIndicatorAndEnableInteraction() {
+        if (mActivityIndicator == null
+                || mActivityIndicatorBackground == null) {
+            return;
+        }
+        mActivityIndicatorBackground.setVisibility(View.INVISIBLE);
+        mActivityIndicatorBackground.setClickable(false);
+        mActivityIndicator.setVisibility(View.INVISIBLE);
+        enableInteraction();
+    }
+
+    private void enableInteraction() {
+        if (mCameraPreview == null
+                || mButtonImportDocument == null
+                || mButtonCameraTrigger == null) {
+            return;
+        }
+        mCameraPreview.setEnabled(true);
+        mButtonImportDocument.setEnabled(true);
+        mButtonCameraTrigger.setEnabled(true);
+    }
+
+    private void disableInteraction() {
+        if (mCameraPreview == null
+                || mButtonImportDocument == null
+                || mButtonCameraTrigger == null) {
+            return;
+        }
+        mCameraPreview.setEnabled(false);
+        mButtonImportDocument.setEnabled(false);
+        mButtonCameraTrigger.setEnabled(false);
+    }
```

Comment:
I used to do methods with flag arguments, but then Rico pointed me to a section in Clean Code (he literally placed the book on my desk with a pan at the Flag Arguments section 😁): 
> Flag arguments are ugly. Passing a boolean into a function is a truly terrible practice. It
immediately complicates the signature of the method, loudly proclaiming that this function does more than one thing. It does one thing if the flag is true and another if the flag is false!
> 
> In Listing 3-7 we had no choice because the callers were already passing that flag
in, and I wanted to limit the scope of refactoring to the function and below. Still, the
method call render(true) is just plain confusing to a poor reader. Mousing over the call
and seeing render(boolean isSuite) helps a little, but not that much. We should have
split the function into two: renderForSuite() and renderForSingleTest().

It's a matter of preference for state toggling methods. I prefer two clear verbose methods. The thing about forgetting to add it to both methods can happen anyhow in case you have to do more than just pass a boolean to the view... and when having to toggle the state of many views we can add them to a list for example.